hydrosdk/modelversion.py

```diff
@@ -535,11 +593,80 @@ class NotFound(Exception):
     class BadRequest(Exception):
         pass
 
-    class BadResponse(Exception):
+
+class DataProfileStatus(Enum):
+    Success = "Success"
+    Failure = "Failure"
+    Processing = "Processing"
+    NotRegistered = "NotRegistered"
+
+
+class DataUploadResponse:
+    """
+    Class that wraps processing status of the training data upload.
+
+    Check the status of the processing using `get_status()`
+    Wait for the processing using `wait()`
+    """
+
+    def __init__(self, cluster: Cluster, model_version_id: int) -> 'DataUploadResponse':
+        self.cluster = cluster
+        self.model_version_id = model_version_id
+
+    @property
+    def url(self) -> str:
+        if not hasattr(self, '__url'):
+            self.__url = '/monitoring/profiles/batch/{}/status'.format(self.model_version_id)
+        return self.__url
+    
+    @staticmethod
+    def __tick(retry: int, sleep: int) -> Tuple[bool, int]:
+        if retry == 0:
+            return False, retry
+        retry -= 1
+        time.sleep(sleep)
+        return True, retry
+
+    def get_status(self) -> DataProfileStatus:
+        response = self.cluster.request('GET', self.url)
+        if response.status_code != 200:
+            raise Cluster.BadResponse('Could not fetch the status of the data processing task')
+        return DataProfileStatus[response.json()['kind']]
+
+    def wait(self, retry=12, sleep=30):
+        """
+        Wait till data processing gets finished. 
+        
+        :param retry: Number of retries before giving up waiting.
+        :param sleep: Sleep interval between retries.
+        """
+        while True:
+            status = self.get_status()
+            if status == DataProfileStatus.Success:
+                break
+            elif status == DataProfileStatus.Processing:
+                can_be_polled, retry = self.__tick(retry, sleep)
+                if can_be_polled:
+                    continue
+                raise DataUploadResponse.DataProcessingNotFinished
+            elif status == DataProfileStatus.NotRegistered:
+                can_be_polled, retry = self.__tick(retry, sleep)
+                if can_be_polled:
+                    continue
+                raise DataUploadResponse.DataProcessingNotRegistered
+            raise DataUploadResponse.DataProcessingFailed
+
+    class DataProcessingNotRegistered(Exception):
+        pass
+
+    class DataProcessingNotFinished(Exception):
+        pass
+
+    class DataProcessingFailed(Exception):
         pass
 
 
-class UploadResponse:
+class ModelUploadResponse:
```

Comment:
Personally, I am in favor of this renaming. Despite extending a name by a few characters it prevents any possible misunderstanding.

Reply:
@Valenzione it creates ambiguity. At the very least there should be a hint in the docstring. Otherwise imho the name should be associated either with localModel or modelVersion as we don't have a model entity

Reply:
Ah, yeah, I missed that point, I got focused on `Why renaming?`. I agree with the ambiguity mentioned.

Reply:
I've introduced a new DataUploadResponse class, which means having UploadResponse might be confusing. ModelUploadResponse makes it more verbose

Reply:
@tidylobster it's just that we don't have a Model entity and using it anywhere makes an ambiguity...